packages/cysync-core/src/actions/wallet/syncWallets.ts

```diff
@@ -34,8 +44,48 @@ export const syncWalletsWithDevice = createAsyncThunk<
   void,
   IDeviceConnectionInfo | undefined,
   { state: RootState }
->('wallets/syncWithDevice', async (conenction, { dispatch }) => {
-  const deletedWallets = await syncWalletsWithConnectedDevice(conenction);
+>('wallets/syncWithDevice', async (connection, { dispatch }) => {
+  const deletedWallets = await syncWalletsWithConnectedDevice(connection);
 
   dispatch(setDeletedWallets(deletedWallets ?? []));
 });
+
+export const fetchWalletsAndSync = createAsyncThunk<
```

Comment:
Add states for `loading`, `success` and `failed` which we can display on the sidebar. Check `deleteWallets` for implementation of loading states.